src/t_zset.c

```diff
@@ -121,7 +121,7 @@ void zslFree(zskiplist *zsl) {
  * levels are less likely to be returned. */
 int zslRandomLevel(void) {
     int level = 1;
-    while ((random()&0xFFFF) < (ZSKIPLIST_P * 0xFFFF))
+    while (level < ZSKIPLIST_MAXLEVEL && (random()&0xFFFF) < (ZSKIPLIST_P * 0xFFFF))
```

Comment:
```suggestion
    while ((random()&0xFFFF) < (ZSKIPLIST_P * 0xFFFF) && level < ZSKIPLIST_MAXLEVEL)
```
Because the probability of the second condition is very low.